spec/ordinary-and-exotic-object-behaviours.html

```diff
@@ -0,0 +1,326 @@
+<!DOCTYPE html>
+<meta charset="utf8" />
+
+<emu-clause id="sec-ordinary-and-exotic-objects-behaviours">
+  <h1>Ordinary and Exotic Objects Behaviours</h1>
+
+  <emu-clause id="sec-built-in-exotic-object-internal-methods-and-slots">
+    <h1>Built-in Exotic Object Internal Methods and Slots</h1>
+    <emu-clause id="sec-record-exotic-objects">
+      <h1>Record Exotic Objects</h1>
+      <p>A Record object is an exotic object that encapsuates a Record value and exposes virtual string-indexed data properties corresponding to the individual keys set on the underlying Record value. All keys properties are non-writable and non-configurable.</p>
+
+      <p>An object is a <dfn id="record-exotic-object">Record exotic object</dfn> (or simply, a Record object) if its following internal methods use the following implementations. These methods are installed in RecordCreate.</p>
+
+      <p>Record exotic objects have the same internal slots as ordinary objects. They also have a [[RecordData]] internal slot.</p>
+
+      <emu-clause id="sec-record-exotic-objects-setprototypeof-v">
+        <h1>[[SetPrototypeOf]] ( _V_ )</h1>
+        <p>When the [[SetPrototypeOf]] internal method of a Record exotic object _R_ is called with argument _V_, the following steps are taken:</p>
+        <emu-alg>
+          1. Return ! SetImmutablePrototype(_R_, _V_).
+        </emu-alg>
+      </emu-clause>
```

Comment:
I was thinking this should follow what other exotic objects do, of defining this as a method, see https://tc39.es/ecma262/#sec-module-namespace-exotic-objects-setprototypeof-v . I'm not really sure which wording to use from what you're quoting, since it doesn't have the internal methods defined for ordinary objects.